### PR TITLE
fix(reactotron-core-client): Fixes `BigInt()` serialization

### LINKED ISSUE #1436: Reactotron does not support `BigInt()`
A community member raised an issue [in the reactotron channel](https://infiniteredcommunity.slack.com/archives/C42C5UKHV/p1707258028882889) that the library doesn't support `BigInt()`.

Indeed, it does not.

```tsx
<Button
  text="BigInt(0)"
  onPress={() => {
    console.log(BigInt(0))
  }}
/>
```

image

### PR DESCRIPTION
**Pull request by markrickert**
Closes #1436 with tests!

Link to community slack: https://infiniteredcommunity.slack.com/archives/C42C5UKHV/p1707258028882889

### CODE REVIEW

**Review by robinheinze — APPROVED** [review]
🚀 LGTM!

**Comment by markrickert** [comment]
The original reporter indicated that this fixes the problem. I don't see any unforeseen issues with adding this to `reactotron-core-client`

**Comment by vasylnahuliak** [comment]
Still reproduce for React Native (only for Android) https://github.com/infinitered/reactotron/issues/1436

**Comment by MisterCommit** [comment]
The issue is not fixed.  @markrickert 

[3.3.14](https://www.npmjs.com/package/reactotron-react-js/v/3.3.14)

[5.1.6](https://www.npmjs.com/package/reactotron-react-native/v/5.1.6)


The same issue is in [serialize.ts](https://github.com/infinitered/reactotron/pull/1437/files#diff-703dd5fd4a52fc95d0621ac1ba3f32e5496fd67969c7fbd28a1f98a5c697b577)

image

**Comment by MisterCommit** [comment]
`
    "@expo/config-plugins": "~7.8.0",
    "@expo/vector-icons": "^14.0.0",
    "@hookform/resolvers": "^1.1.2",
    "@react-native-async-storage/async-storage": "1.21.0",
    "@react-native-community/datetimepicker": "7.6.1",
    "@react-native-community/picker": "^1.8.1",
    "@react-native-community/segmented-control": "2.2.1",
    "@react-navigation/bottom-tabs": "^5.11.10",
    "@react-navigation/drawer": "^5.12.5",
    "@react-navigation/native": "^6.0.2",
    "@react-navigation/stack": "^5.14.4",
    "@types/ramda": "^0.28.23",
    "@types/use-set-timeout": "^1.0.0",
    "@typescript-eslint/parser": "^4.12.0",
    "axios": "^1.6.2",
    "dayjs": "^1.9.7",
    "deepdash-es": "^5.3.9",
    "deprecated-react-native-prop-types": "^2.3.0",
    "expo": "^50.0.14",
    "expo-apple-authentication": "~6.3.0",
    "expo-application": "~5.8.3",
    "expo-asset": "~9.0.2",
    "expo-av": "~13.10.5",
    "expo-blur": "~12.9.2",
    "expo-build-properties": "~0.11.1",
    "expo-constants": "~15.4.5",
    "expo-device": "~5.9.3",
    "expo-document-picker": "~11.10.1",
    "expo-file-system": "~16.0.8",
    "expo-font": "~11.10.3",
    "expo-image-manipulator": "~11.8.0",
    "expo-image-picker": "~14.7.1",
    "expo-jwt": "^1.3.0",
    "expo-linear-gradient": "~12.7.2",
    "expo-linking": "~4.0.1",
    "expo-media-library": "~15.9.1",
    "expo-notifications": "~0.27.6",
    "expo-sharing": "~11.10.0",
    "expo-splash-screen": "~0.26.4",
    "expo-status-bar": "~1.11.1",
    "expo-updates": "~0.24.12",
    "expo-web-browser": "~12.8.2",
    "firebase": "^9.17.2",
    "fuzzy-search": "^3.2.1",
    "global": "^4.4.0",
    "jwt-decode": "^3.1.2",
    "lodash-es": "^4.17.15",
    "lottie-react-native": "6.5.1",
    "markdown-it": "^12.0.4",
    "mathjs": "^9.2.0",
    "metro-react-native-babel-preset": "^0.62.0",
    "patch-package": "^6.4.7",
    "pod-install": "^0.1.38",
    "ramda": "^0.28.0",
    "react": "18.2.0",
    "react-dom": "18.2.0",
    "react-hook-form": "^6.15.3",
    "react-native": "0.73.6",
    "react-native-action-button": "^2.8.5",
    "react-native-auto-height-image": "^3.2.4",
    "react-native-autoheight-webview": "^1.5.8",
    "react-native-confirmation-code-field": "^6.5.1",
    "react-native-dotenv": "^0.2.0",
    "react-native-expo-image-cache": "^4.1.0",
    "react-native-gallery-swiper": "^1.26.4",
    "react-native-gesture-handler": "~2.14.0",
    "react-native-get-random-values": "~1.8.0",
    "react-native-gifted-chat": "^1.0.4",
    "react-native-keyboard-aware-scroll-view": "^0.9.3",
    "react-native-lottie": "^0.0.1-alpha.2",
    "react-native-markdown-renderer": "^3.2.8",
    "react-native-modal": "^11.6.1",
    "react-native-modal-datetime-picker": "^15.0.0",
    "react-native-modal-selector": "^2.0.3",
    "react-native-pager-view": "^6.3.0",
    "react-native-pell-rich-editor": "^1.8.5",
    "react-native-permissions": "^2.1.5",
    "react-native-phone-number-input": "^2.0.0",
    "react-native-picker-modal-view": "^1.3.2",
    "react-native-reanimated": "~3.6.2",
    "react-native-safe-area-context": "^4.5.0",
    "react-native-screens": "~3.29.0",
    "react-native-smart-gallery": "git+https://github.com/Venpal-App/react-native-smart-gallery.git",
    "react-native-status-bar-height": "^2.6.0",
    "react-native-svg": "14.1.0",
    "react-native-svg-transformer": "^0.14.3",
    "react-native-tab-view": "^3.5.2",
    "react-native-uuid": "^2.0.1",
    "react-native-web": "~0.19.6",
    "react-native-webview": "13.6.4",
    "react-query": "^2.13.0",
    "react-spring": "^8.0.27",
    "reactotron-react-js": "^3.3.14",
    "reactotron-react-native": "^5.1.6",
    "rn-tooltip": "^3.0.2",
    "styled-components": "^5.2.1",
    "use-async-effect": "^2.2.2",
    "use-set-timeout": "^1.0.4",
    "yup": "^0.32.9",
    "zustand": "^3.3.2"
  `

File: lib/reactotron-core-client/src/serialize.ts [hunk 1]
```diff
@@ -15,6 +15,21 @@ const INFINITY = "~~~ Infinity ~~~"
 const NEGATIVE_INFINITY = "~~~ -Infinity ~~~"
 // const NAN = '~~~ NaN ~~~'
 
+/**
+ * Fix BigInt serialization
+ * BigInts are not supported by JSON.stringify.  This is a workaround.
+ * https://github.com/GoogleChromeLabs/jsbi/issues/30#issuecomment-953187833
+ */
+declare global {
```
Comment (MisterCommit) [inline lib/reactotron-core-client/src/serialize.ts:23]: // Extend BigInt.prototype.toJSON if BigInt is available and method is not already defined
if (typeof BigInt !== 'undefined' && !BigInt.prototype.toJSON) {
  BigInt.prototype.toJSON = function () {
    return this.toString() + "n"; // Append 'n' to distinguish BigInt
  };
}

**Pull request merged**